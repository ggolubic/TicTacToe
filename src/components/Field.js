import React from "react";
import styled, { css } from "styled-components";

export const Field = props => {
  return (
    <FieldDiv
      player={props.player}
      onClick={props.handleTurn}
      content={props.content}
    >
      {props.children}
    </FieldDiv>
  );
};

const FieldDiv = styled.div`
  background-color: ${props => {
    if (props.content === null) return "#14bdac";
    else if (props.content === "x") return "#545454";
    else return "#ffd02d";
  }};
  margin: 5px;
  height: 100px;
  width: 100px;
  border-radius: 10%;
  font-size: 5rem;
  text-align: center;
  line-height: 75px;
  ${props =>
    !props.content &&
    css`
      &:hover {
        background: ${props => (props.player === "x" ? "#545454" : "#ffd02d")};
        transition: 0.3s;
        cursor: pointer;
      }
    `}
`;
