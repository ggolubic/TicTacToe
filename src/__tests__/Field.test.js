import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import "jest-styled-components";

import Field from "../components/Field";

Enzyme.configure({ adapter: new Adapter() });

describe("Field component", () => {
  it("Should render correctly with default props", () => {
    const tree = renderer.create(<Field />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render correctly with passed props", () => {
    const props = setup();
    const tree = renderer
      .create(<Field {...props}>{props.content}</Field>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render with the correct styles", () => {
    const { content, ...props } = setup();

    let component = mount(
      <Field content="" {...props}>
        {content}
      </Field>
    );
    //if nobody clicked, render greenish background
    expect(component).toHaveStyleRule("background-color", "#14bdac");

    let componentWithX = mount(
      <Field content="x" {...props}>
        {content}
      </Field>
    );
    //if player x clicked render greyish background
    expect(componentWithX).toHaveStyleRule("background-color", "#545454");

    let componentWithO = mount(
      <Field content="o" {...props}>
        {content}
      </Field>
    );
    //if player o clicked render yellowish background
    expect(componentWithO).toHaveStyleRule("background-color", "#ffd02d");

    component.unmount();
  });

  it("Should call props.onClick when div is clicked", () => {
    const { handleTurn, ...props } = setup();
    const component = mount(
      <Field handleTurn={handleTurn} {...props}>
        ""
      </Field>
    );
    // Test before event
    expect(handleTurn).not.toHaveBeenCalled();

    // simulate the click event
    component
      .find("div")
      .props()
      .onClick();
    // Test after event
    expect(handleTurn).toHaveBeenCalled();

    component.unmount();
  });
});

function setup() {
  const mockFunction = jest.fn();
  const props = {
    player: "x",
    handleTurn: mockFunction,
    content: "x",
    size: 5
  };
  return { ...props };
}
