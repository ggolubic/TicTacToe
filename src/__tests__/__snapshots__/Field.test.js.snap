// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component Should render correctly with default props 1`] = `
.c0 {
  background-color: #14bdac;
  margin: 5px;
  position: relative;
  width: calc(100% / 3 - 10px);
  height: calc(100% / 3 - 15px);
  border-radius: 10%;
  text-align: center;
}

.c0:hover {
  background: #545454;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c0 h2 {
  position: absolute;
  top: 30%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin: 0 auto;
  padding: 0;
  font-size: 5vw;
}

@media (max-width:1000px) {
  .c0 h2 {
    top: 40%;
  }
}

<div
  className="c0"
  content=""
  onClick={[Function]}
  size={3}
>
  <h2 />
</div>
`;

exports[`Field component Should render correctly with passed props 1`] = `
.c0 {
  background-color: #545454;
  margin: 5px;
  position: relative;
  width: calc(100% / 5 - 10px);
  height: calc(100% / 5 - 15px);
  border-radius: 10%;
  text-align: center;
}

.c0 h2 {
  position: absolute;
  top: 30%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin: 0 auto;
  padding: 0;
  font-size: 5vw;
}

@media (max-width:1000px) {
  .c0 h2 {
    top: 40%;
  }
}

<div
  className="c0"
  content="x"
  onClick={[MockFunction]}
  size={5}
>
  <h2>
    x
  </h2>
</div>
`;
